### PR TITLE
[SYCL][CUDA] Fix incorrectly dispatched PI methods

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

**Comment by smaslov-intel** [comment]
@alexbatashev , can you add a test?

File: sycl/test/regression/cuda_program_interop.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+int main() {
+  sycl::queue Queue{sycl::default_selector{}};
+  Queue.submit(
+      [](sycl::handler &CGH) { CGH.single_task<class TestKernel>([] {}); });
+
+  sycl::program Prog{Queue.get_context()};
+
+  auto NativeProgram = Prog.get_native<sycl::backend::cuda>();
+
```
Comment (romanovvlad) [inline sycl/test/regression/cuda_program_interop.cpp:16]: Could you please also create a program from the native handle to test piextProgramCreateWithNativeHandle ?

Comment (alexbatashev) [reply sycl/test/regression/cuda_program_interop.cpp:16]: I'd love to, but I only see implementations of `make_platform` for L0 and OpenCL backends, not for CUDA. That's probably why we haven't face that issue before. I only found this by staring at the code.

Comment (romanovvlad) [reply sycl/test/regression/cuda_program_interop.cpp:16]: Ok.

**Review by romanovvlad — DISMISSED** [review]

**Comment by romanovvlad** [comment]
The issue on windows is a known one and should be fixed soon.

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**